src/app/tasks-layout/tasks-layout.component.ts

```diff
@@ -3,39 +3,55 @@ import { HttpClient } from '@angular/common/http';
 import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
 import { EditTaskComponent } from '../user/edit-task/edit-task.component';
 import { ReloadService } from '../reload.service';
+import {SearchService} from '../search.service';
+import { convertActionBinding } from '@angular/compiler/src/compiler_util/expression_converter';
 
 @Component({
   selector: 'app-tasks-layout',
   templateUrl: './tasks-layout.component.html',
   styleUrls: ['./tasks-layout.component.scss']
 })
 export class TasksLayoutComponent implements OnInit {
+
   tasksToday: any[] = [];
   tasksUpcoming: any[] = [];
   tasksCompleted: any[] = [];
+  searchResults: any[] = [];
+  H1: string;
+  H2: string;
+  H3: string;
+  H4: string;
 
   @Input() heading: Text;
   color1 = '#8bd136';
   color2 = '#33e0f1';
-  H1 = 'Due Today';
-  H2 = 'Upcoming';
-  H3 = 'Completed';
+  
+
+  constructor(private data: SearchService,private http: HttpClient, private dialogBox: MatDialog, private reload: ReloadService) {
 
-  constructor(private http: HttpClient, private dialogBox: MatDialog, private reload: ReloadService) { }
+   }
 
 
   ngOnInit(): void {
+    this.data.currentMessage.subscribe(searchResults => this.searchResults = searchResults);
     this.reload.action.subscribe((op) => {
-      this.getAllTasks();
+        console.log(this.searchResults.length);
+        this.getAllTasks();
     });
   }
 
   getAllTasks() {
+
+    // if(this.searchResults.length == 0){
+
     this.http.get('http://localhost:3000/allTasks', { responseType: 'json' }).subscribe(
       (response: any[]) => {
         this.tasksToday = [];
         this.tasksUpcoming = [];
         this.tasksCompleted = [];
+         this.H1 = 'Due Today';
+         this.H2 = 'Upcoming';
+         this.H3 = 'Completed';
```

Comment:
where's H4?

Reply:
Search results

Reply:
"Where" have you assigned H4 its value? I see the said assignment as commented code just below.

Reply:
It's not working, that's why it's commented.Imma try to fix it today